src/ini.zig

```diff
@@ -0,0 +1,1076 @@
+                        // [foo]
+                        // hello = 420
+                        // ```
+                        //
+                        // In the above case, `this.out[section]` would be a string.
+                        // So what should we do in that case?
+                        //
+                        // npm/ini's will chug along happily trying to assign keys to the string.
+                        //
+                        // In JS assigning keys to string does nothing.
+                        //
+                        // Technically, this would have an effect if the value was an array:
+                        //
+                        // ```ini
+                        // foo[] = 0
+                        // foo[] = 1
+                        //
+                        // [foo]
+                        // 0 = 420
+                        // ```
+                        //
+                        // This would result in `foo` being `[420, 1]`.
+                        //
+                        // To be honest this is kind of crazy behavior so we're just going to skip this for now.
+                        skip_until_next_section = true;
+                        continue;
+                    },
+                };
+                head = parent_object.data.e_object;
+                continue;
+            }
+            if (skip_until_next_section) continue;
+
+            // Otherwise it's a key val here
+
+            const line_offset: i32 = @intCast(@intFromPtr(line.ptr) - @intFromPtr(this.src.ptr));
+
+            const maybe_eq_sign_idx = std.mem.indexOfScalar(u8, line, '=');
+
+            const key_raw: []const u8 = try this.prepareStr(ropealloc, line[0 .. maybe_eq_sign_idx orelse line.len], line_offset, .key);
+            const is_array: bool = brk: {
+                if (this.opts.bracked_array) {
+                    break :brk key_raw.len > 2 and bun.strings.endsWith(key_raw, "[]");
+                } else {
+                    // const gop = try duplicates.getOrPut(allocator, key_raw);
+                    // if (gop.found_existing) {
+                    //     gop.value_ptr.* = 1;
+                    // } else gop.value_ptr.* += 1;
+                    // break :brk gop.value_ptr.* > 1;
+                    @panic("We don't support this right now");
+                }
+            };
+
+            const key = if (is_array and bun.strings.endsWith(key_raw, "[]"))
+                key_raw[0 .. key_raw.len - 2]
+            else
+                key_raw;
+
+            if (bun.strings.eql(key, "__proto__")) continue;
+
+            const value_raw: Expr = brk: {
+                if (maybe_eq_sign_idx) |eq_sign_idx| {
+                    if (eq_sign_idx + 1 < line.len) break :brk try this.prepareStr(
+                        ropealloc,
+                        line[eq_sign_idx + 1 ..],
+                        @intCast(line_offset + @as(i32, @intCast(eq_sign_idx)) + 1),
```

Comment:
I don't feel good about all these intcasts and adding up numbers without using slices